src/plugins/visualize/public/application/components/visualize_editor_common.tsx

```diff
@@ -53,6 +57,55 @@ export const VisualizeEditorCommon = ({
   embeddableId,
   visEditorRef,
 }: VisualizeEditorCommonProps) => {
+  const { services } = useKibana<VisualizeServices>();
+
+  useEffect(() => {
+    async function aliasMatchRedirect() {
+      const sharingSavedObjectProps = visInstance?.savedVis.sharingSavedObjectProps;
+      if (services.spaces && sharingSavedObjectProps?.outcome === 'aliasMatch') {
+        // We found this object by a legacy URL alias from its old ID; redirect the user to the page with its new ID, preserving any URL hash
+        const newObjectId = sharingSavedObjectProps?.aliasTargetId; // This is always defined if outcome === 'aliasMatch'
+        const newPath = `${getFullPath(newObjectId!)}${services.history.location.search}`;
```

Comment:
I think this comment also relevant for that place too https://github.com/elastic/kibana/pull/112983#discussion_r721557156

Reply:
done